fwdctl/src/dump.rs

```diff
@@ -24,50 +28,230 @@ pub struct Options {
+        u8_to_string(key)
+    };
+    let value_str = if hex {
+        hex::encode(value)
+    } else {
+        u8_to_string(value)
+    };
+    (key_str, value_str)
+}
+
+async fn handle_next_key(next_key: KeyFromStream) {
+    match next_key {
+        Some(Ok((key, _))) => {
+            println!(
+                "Next key is {0}, resume with \"--start-key={0}\".",
+                u8_to_string(&key)
+            );
+        }
+        Some(Err(e)) => {
+            eprintln!("Error occurred while fetching the next key: {}", e);
+        }
+        None => {
+            println!("There is no next key. Data dump completed.");
+        }
+    }
+}
+
+trait OutputHandler {
+    fn handle_record(&mut self, key: &[u8], value: &[u8]);
+    fn flush(&mut self);
+}
+
+struct CsvOutputHandler {
+    writer: csv::Writer<File>,
+    hex: bool,
+}
+
+impl OutputHandler for CsvOutputHandler {
+    fn handle_record(&mut self, key: &[u8], value: &[u8]) {
+        let (key_str, value_str) = key_value_to_string(key, value, self.hex);
+        self.writer
+            .write_record(&[key_str, value_str])
+            .expect("Failed to write CSV record");
+    }
+
+    fn flush(&mut self) {
+        self.writer.flush().expect("Failed to flush CSV writer");
+    }
+}
+
+struct JsonOutputHandler {
+    writer: BufWriter<File>,
+    hex: bool,
+    is_first: bool,
+}
+
+impl JsonOutputHandler {
+    fn new(mut writer: BufWriter<File>, hex: bool) -> JsonOutputHandler {
+        let _ = writer.write("{\n".as_bytes());
+        JsonOutputHandler {
+            writer,
+            hex,
+            is_first: true,
+        }
+    }
+}
+
+impl OutputHandler for JsonOutputHandler {
+    fn handle_record(&mut self, key: &[u8], value: &[u8]) {
+        let (key_str, value_str) = key_value_to_string(key, value, self.hex);
+        if self.is_first {
+            let _ = self
+                .writer
+                .write(format!("  \"{}\": \"{}\"", key_str, value_str).as_bytes());
+            self.is_first = false;
+        } else {
+            let _ = self
+                .writer
+                .write(format!(",\n  \"{}\": \"{}\"", key_str, value_str).as_bytes());
+        }
+    }
+
+    fn flush(&mut self) {
+        let _ = self.writer.write("\n}\n".as_bytes());
+        self.writer
+            .flush()
+            .expect("Failed to flush Json Buf writer");
+    }
+}
+
+struct StdoutOutputHandler {
+    hex: bool,
+}
+
+impl OutputHandler for StdoutOutputHandler {
+    fn handle_record(&mut self, key: &[u8], value: &[u8]) {
+        let (key_str, value_str) = key_value_to_string(key, value, self.hex);
+        println!("'{}': '{}'", key_str, value_str);
+    }
+    fn flush(&mut self) {}
+}
+
+fn create_output_handler(
+    opts: &Options,
+) -> Result<Box<dyn OutputHandler + Send + Sync>, Box<dyn Error>> {
+    let hex = opts.hex;
+    match opts.output_format.as_str() {
+        "csv" => {
+            let file = File::create("dump.csv")?;
```

Comment:
This should not be hardcoded. Just write to stdout, or provide a way to specify the filename.